### PR TITLE
Store block headers

### LINKED ISSUE #9: Store block headers
## Description

There are multiple consensus rules that require past data from previous block headers.
For convenience we will store block headers data entirely. We expect it will help reorg management later.

## Acceptance Criteria

- Block headers data are stored
- Block headers data are accessible by number and by hash

### PR DESCRIPTION
**Pull request by AbdelStark**
closes #9

### CODE REVIEW

File: src/header/library.cairo [hunk 1]
```diff
@@ -15,29 +15,50 @@ from utils.array import arr_eq
 
 struct BlockHeader:
     member version : felt  # 4 bytes
-    member previous : felt*  # 32 bytes
-    member merkle_root : felt*  # 32 bytes
+    member previous : Uint256  # 32 bytes
```
Comment (ofux) [inline src/header/library.cairo:18]:
```suggestion
    member prev_block : Uint256  # 32 bytes
```
Minor change to follow the naming of the doc and bitcoin core. Not a big deal though

File: src/header/library.cairo [hunk 2]
```diff
@@ -99,44 +118,33 @@ namespace BlockHeaderVerifier:
...
         let (prev0) = swap_endianness_64(data[7] * 2 ** 32 + data[8], 8)
         let (prev1) = swap_endianness_64(data[5] * 2 ** 32 + data[6], 8)
         let (prev2) = swap_endianness_64(data[3] * 2 ** 32 + data[4], 8)
         let (prev3) = swap_endianness_64(data[1] * 2 ** 32 + data[2], 8)
-        assert previous[0] = prev0
-        assert previous[1] = prev1
-        assert previous[2] = prev2
-        assert previous[3] = prev3
+
+        local previous : Uint256 = Uint256(
```
Comment (ofux) [inline src/header/library.cairo:128]: I honestly don't know if this is the good way to build the uint256. I guess @tekkac will have more expertise here.

**Review by ofux — APPROVED** [review]

File: src/header/library.cairo (hunk 1 above)
Comment (AbdelStark) [reply src/header/library.cairo:18]: I wanted to keep the PR  with a single purpose. I think we should do a proper separate PR for naming changes to match bitcoin core naming

File: src/header/library.cairo (hunk 2 above)
Comment (AbdelStark) [reply src/header/library.cairo:128]: I actually checked with him. He reads endianness like French that's handy ;)

Comment (ofux) [reply src/header/library.cairo:128]: ahah perfect!

**Pull request merged**